integration/kubernetes/helm-chart/alluxio/templates/master/statefulset.yaml

```diff
@@ -91,20 +94,26 @@ spec:
       serviceAccountName: {{ .Values.master.serviceAccount }}
     {{- else if .Values.serviceAccount }}
       serviceAccountName: {{ .Values.serviceAccount }}
+    {{- else if $needsJournalFormatJob }}
+      serviceAccountName: job-reader-sa
     {{- end }}
       {{- if .Values.imagePullSecrets }}
 {{ include "alluxio.imagePullSecrets" . | indent 6 }}
       {{- end}}
       initContainers:
-      {{- if .Values.journal.format.runFormat}}
-      - name: journal-format
-        image: {{ .Values.image }}:{{ .Values.imageTag }}
-        imagePullPolicy: {{ .Values.imagePullPolicy }}
-        command: ["alluxio","formatJournal"]
-        volumeMounts:
-          - name: alluxio-journal
-            mountPath: /journal
-      {{- end}}
+      {{- if $needsJournalFormatJob }}
+      - name: wait-for-journal-format
+        image: groundnuty/k8s-wait-for:latest
+        imagePullPolicy: IfNotPresent
+        env:
+          # POD_NAME is the StatefulSet name with the indexed suffix
+          - name: POD_NAME
+            valueFrom:
+              fieldRef:
+                fieldPath: metadata.name
+        # manually override entrypoint command in order to evaluate POD_NAME
+        command: ['sh', '-c', 'wait_for.sh "job" "${POD_NAME}-{{ $journalFormatJobNameSuffix }}"']
```

Comment:
Since there is currently not yet an easy way to access the Pod index number for a StatefulSet, we use the name.
- This was the workaround I found in a [k8s issue on this topic](https://github.com/kubernetes/kubernetes/issues/30427)